specs/merge/fork-choice.md

```diff
@@ -32,36 +31,20 @@ This is the modification of the fork choice according to the executable beacon c
 
 ### `ExecutionEngine`
 
-The following methods are added to the `ExecutionEngine` protocol for use in the fork choice:
-
-#### `set_head`
-
-Re-organizes the execution payload chain and corresponding state to make `block_hash` the head.
+*Note*: The `forkchoice_updated` function is added to the `ExecutionEngine` protocol to signal the fork choice updates.
 
 The body of this function is implementation dependent.
-The Consensus API may be used to implement this with an external execution engine.
-
-```python
-def set_head(self: ExecutionEngine, block_hash: Hash32) -> bool:
-    """
-    Returns True if the ``block_hash`` was successfully set as head of the execution payload chain.
-    """
-    ...
-```
+The Engine API may be used to implement it with an external execution engine.
 
-#### `finalize_block`
+#### `forkchoice_updated`
```

Comment:
Does that mean we have to update fork-choice specs to call `forkchoice_updated` explicitly? Or leave it implementation-dependent?

i.e., naively, we can checks if beacon head `pre_head != post_head` in `on_tick`, `on_block`. and `on_attestation`. And then get the `state.latest_execution_payload_header.block_hash` to call `forkchoice_updated`.

Reply:
I think we better defer the decision on when to make this call to implementations